strawberryfields/backends/tfbackend/circuit.py

```diff
@@ -625,21 +625,14 @@ def measure_fock(self, modes, select=None, **kwargs):
             meas_result = ops.unravel_index(
                 sample_tensor, [self._cutoff_dim] * num_reduced_state_modes
             )
-            if not self._batched:
-                meas_result = meas_result[0]  # no batch index, can get rid of first axis
```

Comment:
@thisac, could this be why the batched tests are failing? Removing this if statement will mean that the TF backend will always return a batched dimension in returned samples, even if `batch_size=1`.

My gut feeling is that we should keep the existing behaviour; when `batch_size=1`, the returned tensors should be of shape `(shots, modes)`, and when `batch_dim>1`, they should be of shape `(batch_size, shots, modes)`.

Reply:
Thanks @josh146. That was part of the cause. Some more issues popped up in a few tests; should be fixed now. When batched the `tf` backend returns shape `(batch_size, shots, measured_modes)`.